class_15_sockets/homework/server.py

```diff
@@ -0,0 +1,78 @@
+import pathlib
+import selectors
+import socket
+from collections import defaultdict
+from multiprocessing import Lock
+from multiprocessing.pool import ThreadPool
+
+HOST = "localhost"
+PORT = 9999
+
+selector = selectors.DefaultSelector()
+pool = ThreadPool(4)
+bufs = defaultdict(bytes)
+clients = defaultdict(str)
+lock = Lock()
+
+
+def append_data(data):
+    data = data.decode()
+    pathlib.Path("outputs").mkdir(parents=True, exist_ok=True)
+    with lock:
+        print("writing to a file")
+        with open(f"outputs/output.txt", "a", encoding="utf-8") as f:
+            f.write(data)
+            f.write("\n")
+
+
+def server(host, port):
+    serv_sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    serv_sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    serv_sock.bind((host, port))
+    serv_sock.listen(5)
+    serv_sock.setblocking(False)
+    selector.register(fileobj=serv_sock, events=selectors.EVENT_READ, data=accept_conn)
```

Comment:
Good scenario for using selectors https://docs.python.org/3/library/selectors.html 👍

Reply:
thank you